CONTRIBUTING.md

```diff
@@ -1,54 +1,124 @@
-- [NodeNetworkConfigurationPolicy Controller](docs/developer-guide-policy.md) - developer guide and design of `NodeNetworkConfigurationPolicy` handler.
+## Local Kubernetes cluster
 
-# Development
+See [local virtualized cluster guide](docs/deployment-local-cluster.md) to learn
+how to deploy a cluster that can be used to try kubernetes-nmstate, run e2e
+tests and perform debugging.
+
+## Building
+
+```shell
+# If pkg/apis/ has been changed, run generator to update client code
+make gen-k8s gen-openapi
+
+# Build handler operator (both its binary and docker image)
+make handler
+```
+
+## Testing
+
+```shell
+# Static checking
+make check
+
+# Run unit tests
+make test/unit
+
+# Run e2e tests, you need a running k8s/openshift cluster with with kubernets-nmstate running.
+make test/e2e
+
+# Run tests matching the regex "NodeSelector"
+make test/e2e E2E_TEST_ARGS='-ginkgo.focus=NodeSelector'
+
+# Conversely, exclude tests that match the regex "Simple\ OVS*"
+make test/e2e E2E_TEST_ARGS='--ginkgo.skip="Simple\ OVS*"'
+```
+
+## Publishing containers
+
+```shell
+# Push nmstate-handler container to remote registry
+make push-handler
+```
+
+It is possible to adjust the built container images with the following
+environment variables.
+
+```shell
+IMAGE_REGISTRY # quay.io
+IMAGE_REPO # nmstate
+
+HANDLER_IMAGE_NAME # kubernetes-nmstate-handler
+HANDLER_IMAGE_TAG # latest
+```
+
+## Manifests
+
+The operator `operator.yaml` manifest from the `deploy` folder is a template to
+be able to replace the with correct docker image to use.
+
+Everytime cluster-sync is called it will regenerate the operator yaml with
+correct kubernets-nmstate-handler image and apply it.
 
-[Development guide](docs/developer-guide-commands.md) is a go-to reference point for
-development helper commands, building, testing, container images and local
-cluster.
 
 # Open a Pull Request
-Kubernetes-nmstate generally follows the [standard github pull request process](https://gist.github.com/Chaser324/ce0505fbed06b947d962), but there is a layer of additional specific differences:
 
-The first difference you'll see is that a bot will begin applying structured labels to your PR.
+kubernetes-nmstate generally follows the [standard github pull request
+process](https://gist.github.com/Chaser324/ce0505fbed06b947d962), but there is a
+layer of additional specific differences:
 
-The bot may also make some helpful suggestions for commands to run in your PR to facilitate review.
-These /command options can be entered in comments to trigger auto-labeling and notifications. Refer to its command reference documentation.
+The first difference you'll see is that a bot will begin applying structured
+labels to your PR.
+
+The bot may also make some helpful suggestions for commands to run in your PR to
+facilitate review. These `/command` options can be entered in comments to
+trigger auto-labeling and notifications. Refer to its command reference
+documentation.
```

Comment:
Include bot commands list [1] also prow PR process  [2]

[1] https://go.k8s.io/bot-commands?repo=nmstate%2Fkubernetes-nmstate
[2] https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process

Reply:
Not a part of this PR, I'm just aligning the existing documentation in a narrow block, breaking lines.